addons/website_slides/static/src/js/slides_course_quiz_manager.js

```diff
@@ -0,0 +1,288 @@
+            if (this._isValidForm(form)) {
+                var values = this._serializeForm(form);
+                return this._rpc({
+                    route: '/slides/slide/quiz/update',
+                    params: values
+                }).then(function (response) {
+                    response.$elem = self.$el;
+                    response.sequence = values.sequence;
+                    self.trigger_up('displayUpdatedQuestion', response);
+                }, function (error) {
+                    self._alertShow(form, error.message.data.arguments[0]);
+                });
+            } else {
+                this._alertShow(form, _t('Please fill up the question'));
+                this.$el.find('.o_wslides_quiz_question input').focus();
+            }
+        },
+
+        _createQuestion: function (createNext) {
+            var self = this;
+            var form = this.$('form');
+            if (this._isValidForm(form)) {
+                var values = this._serializeForm(form);
+                console.log(values);
+                return this._rpc({
+                    route: '/slides/slide/quiz/create',
+                    params: values
+                }).then(function (response) {
+                    response.createNext = createNext;
+                    self.trigger_up('displayCreatedQuestion', response);
+                }, function (error) {
+                    self._alertShow(form, error.message.data.arguments[0]);
+                });
+            } else {
+                this._alertShow(form, _t('Please fill up the question'));
+                this.$el.find('.o_wslides_quiz_question input').focus();
+            }
+        },
+
+        _isValidForm: function(form) {
+            return form.find('.o_wslides_quiz_question input[type=text]').val().trim() !== "";
+        },
+
+        _serializeForm: function (form) {
+            var self = this;
+            var answerIds = [];
+            var answers = [];
+            var sequence = 1;
+            form.find('.o_wslides_js_quiz_answer').each(function () {
+                var value = $(this).find('input[type=text]').val();
+                var answer = {
+                    'sequence': sequence++,
+                    'text_value': value,
+                    'is_correct': $(this).find('input[type=radio]').prop('checked') === true
+                };
+                if (self.update && $(this).data('answer-id')) {
+                    if (value.trim() !== "")
+                        answerIds.push($(this).data('answer-id'));
+                    answer.id = $(this).data('answer-id');
+                    answer.action = 'update';
+                }
+                if (value.trim() !== "")
+                    answers.push(answer);
+            });
+            if (this.update)
+                this.question.answer_ids.forEach(function (id) {
+                    if (!answerIds.includes(id))
+                        answers.push({
```

Comment:
even if there is only one line under the if (or else), we prefer to always use { }, especially here :)